docs/source/docker/advanced/datasource_predictions.rst

```diff
@@ -0,0 +1,347 @@
+you have full control over them and they scale well to millions of samples.
+Furthermore, if you add new samples to your datasource, you can simultaneously
+add their predictions to the datasource.
+If you already have labels instead of predictions, you can treat them
+just as predictions and upload them the same way.
+
+
+Predictions Folder Structure
+----------------------------
+
+In the following, we will outline the format of the predictions required by the
+Lightly docker. Everything regarding predictions will take place in a subdirectory
+of your configured datasource called `.lightly/predictions`. The general structure
+of this directory will look like this:
+
+
+.. code-block:: bash
+
+    datasource/my_dataset
+        + .lightly/predictions/
+            + tasks.json
+            + task_1/
+                 + schema.json
+                 + image_1.json
+                 ...
+                 + image_N.json
+            + task_2/
+                 + schema.json
+                 + image_1.json
+                 ...
+                 + image_N.json
+        + image_1.png
+        + image_2.png
+        + ...
+        + image_N.png
+
+
+Where each subdirectory corresponds to one prediction task (e.g. a classification task
+and an object detection task). All of the files are explained in the next sections.
+
+
+Prediction Tasks
+----------------
+To let Lightly know what kind of prediction tasks you want to work with, Lightly
+needs to know their names. It's very easy to let Lightly know which tasks exist:
+simply add a `tasks.json`` in your storage bucket stored at the subdirectory `.lightly/predictions/`.
+
+The `tasks.json` file must include a list of your task names which must match name
+of the subdirectory where your prediction schemas will be located.
+
+.. note::
+
+    Only the task names listed within `tasks.json`` will be considered.
+    Please ensure that the task name corresponds with the location of your prediction schema.
+    This allows you to specify which subfolder are considered by the Lightly docker.
+
+For example, let's say we are working with the following folder structure:
+
+.. code-block:: bash
+
+    .lightly/predictions/
+        + tasks.json
+        + classification_weather/
+             + schema.json
+             ...
+        + classification_scenery/
+             + schema.json
+             ...
+        + object_detection_people/
+            + schema.json
+            ...
+        + some_directory_containing_irrelevant_things/
+
+
+we can specify which subfolders contain relevant predictions in the `tasks.json`:
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/tasks.json
+
+    [
+        "classification_weather",
+        "classification_scenery",
+        "object_detection_people"
+    ]
+
+.. note::
+
+    If you list a subfolder which doesn't contain a valid `schema.json` file,
+    the Lightly docker will fail! See below how to create a good `schema.json` file.
```

Comment:
Would not say "will fail". That sounds like we did a very bad job. Would rather say "the Lightly docker might not work properly"